### ISSUE TITLE
homectl: don't require user pw when updating record when using unencrypted backend. Also, add update operation that doesn't propgate user record down

### ISSUE THREAD

**Issue by LockRing** [systemd/systemd#16166]
**systemd version the issue has been seen with**
> 245.6-6

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > `homectl update [USER] ...` don't require the password for target user if it runs as root.

**Unexpected behaviour you saw**
 > `homectl update [USER] ...` requires the password for target user even if it runs as root.

**Steps to reproduce the problem**
```
# homectl create test_user --storage=directory
Please enter new password for user test_user: ****                    
Please enter new password for user test_user (repeat): ****
# homectl update test_user --locked=true
Please enter password for user test_user: (press TAB for no echo)
# homectl update test_user --password-change-now=true
Please enter password for user test_user: (press TAB for no echo)
```
`homectl update` requires the user password so administrator can't lock the login or force change password on next login if they don't know the user's password.
Is this an expected behavior? If so is there any way to use --locked and --password-change-now options without user password? I think root should be able to update user records without target user authentication.

**Comment by poettering** [comment]
So we currently propagate the user records all the way down into the home image, unconditionally. That's necessary so that you can take the image elsewhere and the settings apply there too. And since home directories are encrypted using the user's password, you need to provide the user's password that we can do that.

Now, in your case you use a plain directory for storage, i.e. no encryption. I guess for this specific case we can drop the requirement to authenticate properly, since we can make the changes anyway.

We could also add a mode where you can change the record without authentication, in which case only the host copy of the user record is changed, but it is not attempted to change the copy embedded in the home directory, but this would be done on next log-in. This has been on the TODO list already, but so far wasn't implemented

**Comment by LockRing** [comment]
Thanks for the comment. According to your comment, there's no way to control (except remove and passwd under limited conditions) systemd-homed based user if administrator doesn't know the user password.

I think it would be nice if there was data stored only on the machine, not in home image, aside from .identity file. It can be used to block login or set resource management properties only on that machine without a user password. AFAIK systemd-homed allows user login if .public file is saved in /var/lib/systemd/home/ which was used to create that user. Therefore, it can also be used to restrict portable users.

**Comment by poettering** [comment]
> Thanks for the comment. According to your comment, there's no way to control (except remove and passwd under limited conditions) systemd-homed based user if administrator doesn't know the user password.

homectl doesn't expose it. You can patch the user records in /var/lib/systemd/home/, and properly bump their lastChange time. If you do so, the new records take effect and will already be propagated into the homedir itself. This is not exposed via homectl however, we only expose the "update all the way" operation and nothing else.

Maybe we should just change how we operate: when a home is mounted continue to propagate all the way as before, but if not, maybe we should just update the host version and entirely rely on the automatic propagation of updated records at login time.

**Comment by LockRing** [comment]
>Maybe we should just change how we operate: when a home is mounted continue to propagate all the way as before, but if not, maybe we should just update the host version and entirely rely on the automatic propagation of updated records at login time.

I think this is appropriate. It can update user record without the password.

I tried to test portable USB home but `homectl` state shows `unfixated` and `homectl activate` returns `Key has been revoked`. Looks like related to #15178. So I have a question. Does systemd-homed also creates host version and save it permanent when user home USB inserted? If not, that solution can't use for portable home.

**Comment by poettering** [comment]
Addressed by d94c7eef121dd5dbab758722a199125d49f85d55
